Created Proposal for CONTRIBUTING.md (#49)
alumnium-hq/alumnium: CONTRIBUTING.md

> # Contributing to Aluminium

Reviewer: ```suggestion
# Contributing to Alumnium
```

## Welcome to the Future of Test Automation!

> Thank you for your interest in contributing to Aluminium! As an experimental AI-powered test automation framework, we are pioneering new approaches to make testing more intuitive and robust. Your expertise and passion can help shape this emerging technology.

Reviewer: ```suggestion
Thank you for your interest in contributing to Alumnium! As an experimental AI-powered test automation framework, we are pioneering new approaches to make testing more intuitive and robust. Your expertise and passion can help shape this emerging technology.
```

> As a software tester, I have seen how the right tools can transform quality assurance. Aluminium aims to be that transformative tool by leveraging AI to simplify test interactions and assertions.

Reviewer: ```suggestion
As a software tester, I have seen how the right tools can transform quality assurance. Alumnium aims to be that transformative tool by leveraging AI to simplify test interactions and assertions.
```

## Project Understanding

Before contributing, please review:

> - Our [docs](https://alumnium.ai/docs/) and [README.md](https://github.com/alumnium-hq/alumnium) to understand Aluminium's vision: creating higher-level abstractions for test automation that simplify web page interactions and strengthen assertion mechanisms

Reviewer: ```suggestion
- Our [docs](https://alumnium.ai/docs/) and [README.md](https://github.com/alumnium-hq/alumnium) to understand Alumnium's vision: creating higher-level abstractions for test automation that simplify web page interactions and strengthen assertion mechanisms
```

- Our experimental status - we're in early development and value innovative approaches
- The core functionality that uses natural language processing to interpret testing commands

## Finding Your Contribution Opportunity

- Explore the [open issues](https://github.com/alumnium-hq/alumnium/issues) to find tasks matching your interests
- We will be glad if you help us with :
  - Improving test coverage for edge cases  
  - Enhancing documentation and examples  
  - Exploring more natural language prompts for test generation  
  - Reporting usability issues or unexpected test behavior  

>   - Creating sample projects using Aluminium

Reviewer: ```suggestion
  - Creating sample projects using Alumnium
```

## Contribution Workflow

### 1. Environment Setup

```bash
# Fork and clone the repository

> git clone https://github.com/your-username/aluminium.git
> cd aluminium

Reviewer: ```suggestion
git clone https://github.com/your-username/alumnium.git
cd alumnium
```

# Set up development environment
pipx install poetry
poetry install

```

### 2. Configure AI provider access
Configure access to AI providers as mentioned in [docs](https://alumnium.ai/docs/getting-started/configuration/)

### 3. Development Guidelines

> When working on Aluminium:

Reviewer: ```suggestion
When working on Alumnium:
```

# Contributing to Aluminium

## Welcome to the Future of Test Automation!

Thank you for your interest in contributing to Aluminium! As an experimental AI-powered test automation framework, we are pioneering new approaches to make testing more intuitive and robust. Your expertise and passion can help shape this emerging technology.

As a software tester, I have seen how the right tools can transform quality assurance. Aluminium aims to be that transformative tool by leveraging AI to simplify test interactions and assertions.

## Project Understanding

Before contributing, please review:

> - Our [README.md](https://github.com/automatealchemist/alumnium#) to understand Aluminium's vision: creating higher-level abstractions for test automation that simplify web page interactions and strengthen assertion mechanisms

Reviewer: ```suggestion
- Our [docs](https://alumnium.ai/docs/) and [README.md](https://github.com/alumnium-hq/alumnium) to understand Aluminium's vision: creating higher-level abstractions for test automation that simplify web page interactions and strengthen assertion mechanisms
```

- Our experimental status - we're in early development and value innovative approaches
- The core functionality that uses natural language processing to interpret testing commands

## Finding Your Contribution Opportunity

> - Explore the [open issues](https://github.com/yourusername/aluminium/) to find tasks matching your interests

Reviewer: ```suggestion
- Explore the [open issues](https://github.com/alumnium-hq/alumnium/issues) to find tasks matching your interests
```

- We will be glad if you help us with :
  - Improving test coverage for edge cases  
  - Enhancing documentation and examples  
  - Exploring more natural language prompts for test generation  
  - Reporting usability issues or unexpected test behavior  
  - Creating sample projects using Aluminium

## Contribution Workflow

### 1. Environment Setup

```bash
# Fork and clone the repository
git clone https://github.com/your-username/aluminium.git
cd aluminium

# Set up development environment
pipx install poetry
poetry install

# Configure AI provider access as mentioned in [README.md](https://github.com/automatealchemist/alumnium#)

> ```

Reviewer: I think it may be better to move it as a separate step. A contributor will need to decide which model to use.
```suggestion
### 2. Configure AI provider access

Configure access to AI providers as mentioned in [docs](https://alumnium.ai/docs/getting-started/configuration/)
```

  poetry run pytest             # Pytest examples
  ```

### 3. Pull Request Process

1. **Create a focused branch** for your contribution
2. **Write meaningful commit messages** explaining your changes
3. **Include tests** that verify your contribution works as expected
4. **Update documentation** if you're adding or changing features
5. **Submit your PR** with a clear description of what it accomplishes

## AI-First Testing Philosophy

As contributors to an AI-powered testing tool, we value:

- **Natural language over rigid syntax** - Tests should be readable by non-technical stakeholders
- **Adaptability over brittleness** - Tests should withstand UI changes
- **Intent over implementation** - Focus on what should happen, not how it happens
- **Context awareness** - Testing tools should understand the application under test

## Community Guidelines

- Be respectful and constructive in all interactions
- Share knowledge generously - we're all learning in this emerging field
- Value diverse perspectives - they lead to more robust solutions
- Ask questions when unclear - clarity benefits everyone

## For First-Time Contributors

If you're new to open-source or AI-powered testing:

> 1. Try running the demo and experimenting with the Alumni API

Reviewer: ```suggestion
1. Try running the demo and experimenting with the Alumnium API
```
